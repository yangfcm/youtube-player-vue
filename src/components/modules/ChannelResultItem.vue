<template>
  <div class="ui items">
    <div class="ui item">
      <router-link class="ui small image circular" :to="`/channel/${channel.id.channelId}`">
        <img :src="channel.snippet.thumbnails.medium.url" :title="channel.snippet.channelTitle" />
      </router-link>
      <div class="content">
        <router-link
          class="header"
          :to="`/channel/${channel.id.channelId}`"
        >{{ channel.snippet.channelTitle }}</router-link>
        <div class="description">{{ channel.snippet.description }}</div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ["channel"],
};
</script>
<style scoped>
</style>