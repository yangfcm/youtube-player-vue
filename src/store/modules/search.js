import axios from "../../settings";
import { SEARCH, CATCH_SEARCH_ERROR } from "../types";
import parseError from "../helpers/parseError";

const state = {
  searchResults: null,
  searchError: null,
};

const getters = {
  searchErrorMessage: (state) => {
    return parseError(state.searchError, "Failed to proceed searching");
  },
};

const mutations = {
  SEARCH: (state, payload) => {
    state.searchResults = payload;
    state.searchError = null;
  },
  CATCH_SEARCH_ERROR: (state, payload) => {
    state.searchError = payload;
  },
};

const actions = {
  search: async (context, keyword, pageToken) => {
    try {
      const response = await axios.get("/search", {
        params: {
          ...axios.defaults.params,
          q: keyword,
          part: "snippet",
          maxResults: 15,
          pageToken,
        },
      });
      context.commit(SEARCH, response.data);
    } catch (err) {
      context.commit(CATCH_SEARCH_ERROR, err);
    }
  },
};

export default {
  state,
  getters,
  mutations,
  actions,
};
