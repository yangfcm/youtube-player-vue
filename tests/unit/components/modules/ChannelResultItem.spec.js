import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import ChannelResultItem from "@/components/modules/ChannelResultItem";
import { searchResultResponse } from "../../store/fixtures/search";

describe("Test ChannelResultItem component", () => {
  let wrapper;
  const channel = searchResultResponse.items.find(
    (item) => item.id.kind === "youtube#channel"
  );
  beforeEach(() => {
    wrapper = shallowMount(ChannelResultItem, {
      propsData: {
        channel,
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct content", () => {
    expect(wrapper.find("img").attributes("src")).toBe(
      channel.snippet.thumbnails.medium.url
    );
    expect(wrapper.find(".header").text()).toContain(
      channel.snippet.channelTitle
    );
    expect(wrapper.find("div.description").text()).toContain(
      channel.snippet.description
    );
    expect(wrapper.find("a").props().to).toBe(
      `/channel/${channel.id.channelId}`
    );
  });
});
