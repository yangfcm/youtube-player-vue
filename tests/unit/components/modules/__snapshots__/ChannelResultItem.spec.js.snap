// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChannelResultItem component should match snapshot 1`] = `
<div class="ui items">
  <div class="ui item"><a class="ui small image circular"><img src="https://yt3.ggpht.com/-Fgp8KFpgQqE/AAAAAAAAAAI/AAAAAAAAAAA/Wyh1vV5Up0I/s240-c-k-no-mo-rj-c0xffffff/photo.jpg" title="Google Developers"></a>
    <div class="content"><a class="header">Google Developers</a>
      <div class="description">The Google Developers channel features talks from events, educational series, best practices, tips, and the latest updates across our products and platforms.</div>
    </div>
  </div>
</div>
`;
